site/app/templates/misc/MarkdownArea.twig

```diff
@@ -41,7 +41,8 @@
             placeholder="{{ placeholder }}"
         {% endif %}
         rows="10" cols="30"
-        maxlength="5000"
+        {# 524288 characters is the default max length for text input fields #}
+        maxlength="{{no_maxlength is defined and no_maxlength ? '524288' : '5000'}}"
```

Comment:
Is 5000 chars as the standard not enough? google says that's 1.5 pages single spaced. Just wondering because maybe we shouldn't introduce an new variable just to do this unless it's needed

Reply:
The reason I removed this limit for this page was because 
1) The original page didn't have a limit to the size of solutions
2) If an instructor has a large code snippet that they want to insert as a solution (or maybe they want to show multiple solutions for the same problem), they are able to paste it in the box even with the maxlength=5000, but then they would be unable to type any more characters (only backspace) which would be very confusing.
3) I think the instructor should just be able to specify a solution of any length, unlike the forum where we want to stop students from being able to, say, post their entire homework code.

Do you think these reasons are good? More than happy to discuss further if not. Just to clarify, all other places that use this `MarkdownArea.twig` template right now still have a limit of 5000 characters.

Reply:
Okay sounds goodd! You're right with code snippets maybe it'll exceed 5000 chars

Reply:
Nitpick / up to you, but this could be more concise using `[|default](https://twig.symfony.com/doc/2.x/filters/default.html)` like this: `{{no_maxlength|default(null) ? '524288' : '5000'}}`

Reply:
Oh cool, I wasn't aware of this filter! Thanks for the suggestion.